Library/ViewModels/WatchProjectViewModel.swift

```diff
@@ -171,8 +169,7 @@ public final class WatchProjectViewModel: WatchProjectViewModelType,
     self.saveButtonAccessibilityValue = self.saveButtonSelected
       .map { starred in starred ? Strings.Saved() : Strings.Unsaved() }
 
-    self.postNotificationWithProject = project
-      .takeWhen(saveButtonTapped)
+    self.postNotificationWithProject = projectOnSaveButtonToggleSuccess
```

Comment:
So now we're only firing the notification when the request actually succeeds. Does this have any user experience implications for other parts of the app where the save icon is shown? Was this intentionally done this way previously, perhaps for "optimistic UI" reasons?

Reply:
Using `project` was sometimes causing a behaviour where after un-saving a project, the `Saved` list would not remove the project. Instead, it would simply hide the heart icon. I noticed though that this doesn't happen often. So I guess we could revert this. What do you think?